QPong-Unity-Version/Assets/Scripts/CircuitGridClient.cs

```diff
@@ -27,70 +31,97 @@ void Start()
+            GetStateVector(gateArrayString);
         }
+       
         if (doMeasurementFlag) {
             doMeasurementFlag = false;
-            var gateArrayString = string.Join(",", GameObject.Find("CircuitGrid").GetComponent<CircuitGridControl>().gateArray);
-            StartCoroutine(doMeasurement(gateArrayString));
+            DoMeasurement(gateArrayString);
         }
     }
 
-    IEnumerator getStatevector(string gateArrayString)
+    private void GetStateVector(string gateString)
     {
-        Debug.Log("Send Gate Array: "+ gateArrayString);
-        List<IMultipartFormSection> formData = new List<IMultipartFormSection>();
-        formData.Add(new MultipartFormDataSection("gate_array", gateArrayString));
-        UnityWebRequest www = UnityWebRequest.Post("http://127.0.0.1:8008/api/run/get_statevector", formData);
-        yield return www.SendWebRequest();
-        Debug.Log("Response: " + www.downloadHandler.text);
-        
-        // Deserialize stateVector from JSON
-        var obj = JsonConvert.DeserializeObject<RootObject>(www.downloadHandler.text);
-        Complex[] stateVector = new Complex[numberOfStates];
-        double[] stateProbability = new double[numberOfStates];
-        paddleArray = GameObject.Find("CircuitGrid").GetComponent<CircuitGridControl>().paddleArray;
-        for (int i = 0; i < numberOfStates; i++){
-            stateVector[i] = new Complex(obj.__ndarray__[i].__complex__[0],obj.__ndarray__[i].__complex__[1]);
-            stateProbability[i] = Complex.Pow(stateVector[i], 2).Magnitude;
-            paddleArray[i].GetComponent<SpriteRenderer> ().color = new Color (1,1,1,(float)stateProbability[i]);
-        }
-        Debug.Log("State Probability: ["+string.Join(", ", stateProbability)+"]");
+        Debug.Log("Send Gate Array: "+ gateString);
+        string urlString = API_URL + API_VERSION + "get_statevector";
+        StartCoroutine(PostRequest(urlString, gateString, (results) => {
+
+            // Deserialize stateVector from JSON
+            //TODO: come up with a better way to abstract this out
+            var obj = JsonConvert.DeserializeObject<RootObject>(results);
+            Complex[] stateVector = new Complex[numberOfStates];
+            double[] stateProbability = new double[numberOfStates];
+            paddleArray = GameObject.Find("CircuitGrid").GetComponent<CircuitGridControl>().paddleArray;
+            for (int i = 0; i < numberOfStates; i++)
+            {
+                stateVector[i] = new Complex(obj.__ndarray__[i].__complex__[0], obj.__ndarray__[i].__complex__[1]);
+                stateProbability[i] = Complex.Pow(stateVector[i], 2).Magnitude;
+                paddleArray[i].GetComponent<SpriteRenderer>().color = new Color(1, 1, 1, (float)stateProbability[i]);
+            }
+           // Debug.Log("State Probability: [" + string.Join(", ", stateProbability) + "]");
+        }));
+    }
+
+    IEnumerator DelayStateVector(string gatesString, float delay)
```

Comment:
Should we stick to either "gateString" or "gatesString"?

Reply:
yes we should your right